main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -747,4 +747,29 @@ object WeederError {
       s"${underline("Tip:")} Type parameters are not allowed on effects."
     })
   }
+
+  /**
+    * An error raised to indicate a use of resume outside an effect handler.
+    *
+    * @param loc the location where the error occurred.
+    */
+  case class IllegalResume(loc: SourceLocation) extends WeederError {
+    def summary: String = "Unexpected use of 'resume'."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Unexpected use of 'resume'.
+         |
+         |${code(loc, "unexpected use of 'resume'")}
+         |
+         |""".stripMargin
+    }
+
+    def explain(formatter: Formatter): Option[String] = Some({
+      import formatter._
+      s"${underline("Tip:")} The 'resume' expression may only be used in effect handlers."
```

Comment:
I think we should include the statement "resume must occur in an effect handler" in the summary and message itself, and then leave the explain = None.